Simplenote/src/androidTest/java/com/automattic/simplenote/smoke/test/TestRunner.java

```diff
@@ -0,0 +1,405 @@
+package com.automattic.simplenote.smoke.test;
+
+import androidx.test.rule.ActivityTestRule;
+
+import com.automattic.simplenote.NotesActivity;
+import com.automattic.simplenote.smoke.data.DataProvider;
+import com.automattic.simplenote.smoke.data.NoteDTO;
+import com.automattic.simplenote.smoke.data.SignUpDataProvider;
+import com.automattic.simplenote.smoke.pages.IntroPage;
+import com.automattic.simplenote.smoke.pages.LoginPage;
+import com.automattic.simplenote.smoke.pages.MainPage;
+import com.automattic.simplenote.smoke.pages.SettingsPage.SortOrder;
+import com.automattic.simplenote.smoke.utils.TestUtils;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class TestRunner {
+    @Rule
+    public ActivityTestRule<NotesActivity> mActivityTestRule = new ActivityTestRule<>(NotesActivity.class, true, true);
+
+    private static String email;
+    private static String password;
+
+    @Before
+    public void init() {
+        TestUtils.logoutIfNecessary(DataProvider.UNKNOWN_EMAIL);
+
+        if (email == null || password == null) {
+
+            email = SignUpDataProvider.generateEmail();
+            password = SignUpDataProvider.generatePassword();
+
+            System.out.println("Email: " + email);
+            System.out.println("Password: " + password);
+
+            new IntroPage()
+                    .openSignUp()
+                    .signUp(email, password)
+                    .logout(email);
+        }
+    }
+
+    @After
+    public void tearDown() {
+    }
+
+    @Test
+    public void testLogin() {
+
+        new IntroPage()
+                .goToLoginWithEmail()
+                .login(email, password);
+
+        new MainPage()
+                .isOpened()
+                .logout(email);
+    }
+
+    @Test
+    public void testLogout() {
+
+        if (!TestUtils.logoutIfNecessary(email)) {
+            new IntroPage()
+                    .goToLoginWithEmail()
+                    .login(email, password);
+
+            new MainPage()
+                    .isOpened()
+                    .logout(email);
+        }
+    }
+
+    @Test
+    public void testLoginWithWrongPassword() {
+        TestUtils.logoutIfNecessary(email);
```

Comment:
Is this needed? I'm trying to understand how it's different from the setup in `@Before`.

Reply:
That is a leftover part of an unused approach. It will be removed.